### PR TITLE
GHA: Split scheduled tests into a seperate workflow

### PR DESCRIPTION
**Pull request by m-richards**
This was inspired by the versioneer change to not run on forks. I would quite like to be able to run tests on my fork, but not have the scheduled tests trigger every day (and generate daily notifications when the dev build is broken). Currently I work around this by turning the tests workflow on and off globally under the actions tab:

image

Splitting the workflow conditions means these could be turned on and off individually. Happy though if we decide it's not worth the complexity - I had a quick look and can't see other projects doing this, so perhaps this is only an annoyance for me.

### CODE REVIEW

**Comment by martinfleis** [comment]
Is there a way to run the scheduled tests only on geopandas/geopandas? I am not sure if the `if` can be outside jobs.

**Comment by jorisvandenbossche** [comment]
https://github.com/orgs/community/discussions/26684 might be relevant

**Comment by m-richards** [comment]
Thanks both for the suggestions that's much cleaner

File: .github/workflows/tests.yaml [hunk 1]
```diff
@@ -115,3 +117,6 @@ jobs:
           pytest -v --color=yes --doctest-only geopandas --ignore=geopandas/datasets
 
       - uses: codecov/codecov-action@v4
+
+    # skip scheduled runs on forks
+    if: (github.event_name == 'schedule' && github.repository == 'geopandas/geopandas') || (github.event_name != 'schedule')
```
Comment (martinfleis) [inline .github/workflows/tests.yaml:122]: Can we move it closer to the top of the job definition? It is super hidden here.

**Review by martinfleis — APPROVED** [review]

**Review by jorisvandenbossche — APPROVED** [review]

**Pull request merged**